modules/honzon-distribution/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+			if let Some(capacity) = capacity {
+				params.capacity = capacity;
+			}
+			if let Some(max_step) = max_step {
+				params.max_step = max_step;
+			}
+			if let Some(target_min) = target_min {
+				params.target_min = target_min;
+			}
+			if let Some(target_max) = target_max {
+				params.target_max = target_max;
+			}
+			DistributionDestinationParams::<T>::insert(&destination, &params);
+
+			Self::deposit_event(Event::<T>::UpdateDistributionParams { destination, params });
+
+			Ok(())
+		}
+
+		#[pallet::weight(T::WeightInfo::force_adjust())]
+		#[transactional]
+		pub fn force_adjust(
+			origin: OriginFor<T>,
+			destination: DistributionDestination<T::AccountId>,
+		) -> DispatchResult {
+			T::UpdateOrigin::ensure_origin(origin)?;
+
+			Self::do_adjust_to_destination(destination)?;
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn do_adjust_to_destination(destination: DistributionDestination<T::AccountId>) -> DispatchResult {
+		let params =
+			DistributionDestinationParams::<T>::get(&destination).ok_or(Error::<T>::DistributionParamsNotExist)?;
+		match destination.clone() {
+			DistributionDestination::StableAsset(stable_asset) => {
+				let balance = Self::adjust_for_stable_asset(&destination, stable_asset, params)?;
+
+				// update `DistributedBalance` of destination
+				DistributedBalance::<T>::try_mutate(destination, |maybe_balance| -> DispatchResult {
+					let old_val = maybe_balance.take().unwrap_or_default();
+					let new_val = if balance.is_positive() {
+						old_val.saturating_add(balance as Balance)
+					} else {
+						old_val.saturating_sub(balance.unsigned_abs())
+					};
+					*maybe_balance = Some(new_val);
+
+					Ok(())
+				})?;
+			}
+		}
+		Ok(())
+	}
+
+	/// if current value less than target_min, mint aUSD:
+	///     (balances+x)/(total_supply+x)=target,
+	///     x=(target*total_supply-balances)/(1-target)
+	/// if current value large than target_max, burn aUSD:
+	///     (balances+x)/(total_supply+x)=target,
+	///     x=(balances-target*total_supply)/(1-target)
+	///
+	/// return `Amount` that will be add to or subtract from `DistributedBalance`.
+	fn adjust_for_stable_asset(
+		destination: &DistributionDestination<T::AccountId>,
+		stable_asset: DistributionToStableAsset<T::AccountId>,
+		params: DistributionParams,
+	) -> Result<Amount, DispatchError> {
+		let stable_currency = T::GetStableCurrencyId::get();
+		let pool_id = stable_asset.pool_id;
+		let pool_info = T::StableAsset::pool(pool_id).ok_or(Error::<T>::InvalidDestination)?;
+		let account_id = stable_asset.account_id;
+		let asset_length = pool_info.assets.len();
+		let asset_index = stable_asset.stable_token_index as usize;
+		let total_supply = pool_info.total_supply;
+		let ausd_supply = pool_info
+			.balances
+			.get(asset_index)
+			.ok_or(Error::<T>::InvalidDestination)?;
+		ensure!(asset_index < asset_length, Error::<T>::InvalidDestination);
```

Comment:
I don't think we need this check anymore

Reply:
line #368 would panic if governance accidentally put a bad value.

I think check is still useful